### PR TITLE
add linear API in NLS constructor

### PR DESCRIPTION
**Pull request by tmigot**
This is the NLS-version of https://github.com/JuliaSmoothOptimizers/ADNLPModels.jl/pull/69
Again this is extracted from the large PR https://github.com/JuliaSmoothOptimizers/ADNLPModels.jl/pull/66 (so I refer to #66 for the unit tests)

### CODE REVIEW

File: src/nls.jl [hunk 1]
```diff
@@ -19,15 +26,24 @@ ADNLPModels.show_header(io::IO, nls::ADNLSModel) = println(
...
+    ADNLSModel(F, x0, nequ, lvar, uvar, A, c, lcon, ucon)
 
 ADNLSModel is an Nonlinear Least Squares model using automatic differentiation to
 compute the derivatives.
 The problem is defined as
 
      min  ½‖F(x)‖²
-    s.to  lcon ≤ c(x) ≤ ucon
+    s.to  lcon ≤ (  Ax  ) ≤ ucon
+                 ( c(x) )
```
Comment (dpo) [inline src/nls.jl:46]: Maybe just to make it clear, state that the linear constraints comes first in the list of constraints.

And so, do you still need the keyword `linequ` below?

Comment (dpo) [reply src/nls.jl:46]: On second reading, maybe `linequ` is to indicate linear *residuals*. If so, that could be explained more clearly.

**Review by dpo — APPROVED** [review]

**Comment by dpo** [comment]
Excellent, thank you!

File: src/nls.jl (hunk 1 above)
Comment (tmigot) [reply src/nls.jl:46]: the linear part of the residual is a bit work to be done, so I will had to the todo-list this clarification. Thanks

**Pull request merged**